src/v/pandaproxy/schema_registry/requests/post_subject_versions.h

```diff
@@ -0,0 +1,240 @@
+                                     .match("schema", state::schema)
+                                     .match("schemaType", state::schema_type)
+                                     .match("references", state::references)
+                                     .default_match(std::nullopt)};
+            if (s.has_value()) {
+                _state = *s;
+            }
+            return s.has_value();
+        }
+        case state::reference: {
+            std::optional<state> s{string_switch<std::optional<state>>(sv)
+                                     .match("name", state::reference_name)
+                                     .match("subject", state::reference_subject)
+                                     .match("version", state::reference_version)
+                                     .default_match(std::nullopt)};
+            if (s.has_value()) {
+                _state = *s;
+            }
+            return s.has_value();
+        }
+        case state::empty:
+        case state::schema:
+        case state::schema_type:
+        case state::references:
+        case state::reference_name:
+        case state::reference_subject:
+        case state::reference_version:
+            return false;
+        }
+        return false;
+    }
+
+    bool Uint(int i) {
+        switch (_state) {
+        case state::reference_version:
+            result.references.back().version = schema_version{i};
+            _state = state::reference;
+            return true;
+        case state::empty:
+        case state::record:
+        case state::schema:
+        case state::schema_type:
+        case state::references:
+        case state::reference:
+        case state::reference_name:
+        case state::reference_subject:
+            return false;
+        }
+        return false;
+    }
+
+    bool String(const Ch* str, rapidjson::SizeType len, bool) {
+        auto sv = std::string_view{str, len};
+        switch (_state) {
+        case state::schema: {
+            auto def = make_schema_definition<Encoding>(sv);
+            if (!def) {
+                return false;
+            }
+            result.schema = std::move(def).value();
+            _state = state::record;
+            return true;
+        }
+        case state::schema_type: {
+            std::optional<schema_type> type{
+              string_switch<std::optional<schema_type>>(sv)
+                .match(to_string_view(schema_type::avro), schema_type::avro)
+                .match(to_string_view(schema_type::json), schema_type::json)
+                .match(
+                  to_string_view(schema_type::protobuf), schema_type::protobuf)
+                .default_match(std::nullopt)};
+            if (type.has_value()) {
+                result.type = *type;
+                _state = state::record;
+            }
+            return type.has_value();
+        }
+        case state::reference_name:
+            result.references.back().name = ss::sstring{sv};
```

Comment:
should we be consistent and always add curlies around complex statements on this switch